code/modules/mob/living/silicon/robot/component.dm

```diff
@@ -34,13 +34,15 @@
 	installed = -1
 	uninstall()
 
-/datum/robot_component/proc/take_damage(brute, electronics, sharp, edge)
-	if(installed != 1) return
+/datum/robot_component/proc/take_damage(brute, electronics, sharp)
```

Comment:
this was clearly copy-pasted from  human organ damage at one point; the `sharp` var here isn't even used for ANYTHING.